homeassistant/components/balboa/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Balboa Spa Client integration."""
+import asyncio
+import time
+
+from pybalboa import BalboaSpaWifi
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+import homeassistant.util.dt as dt_util
+
+from .const import _LOGGER, CONF_SYNC_TIME, DEFAULT_SYNC_TIME, DOMAIN, PLATFORMS
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Balboa Spa from a config entry."""
+    host = entry.data[CONF_HOST]
+
+    _LOGGER.debug("Attempting to connect to %s", host)
+    spa = BalboaSpaWifi(host)
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = spa
```

Comment:
Please move this after the `raise ConfigEntryNotReady` so we don't store anything until we know we are going to setup